snapcraft/models/project.py

```diff
@@ -505,64 +723,197 @@ class App(models.CraftBaseModel):
+        default=None,
+        description="The list of slots that the app provides.",
+        examples=["slots: [dbus-daemon]"],
     )
+    """The list of slots that the app provides.
+
+    Slot connections are only made when the snap is running in ``strict``
+    confinement.
+
+    Slots are used to define what code and data can be shared with other snaps.
+
+    See the `content interface <https://snapcraft.io/docs/content-interface>`_
+    for more information about plugs and slots.
+    """
+
     plugs: UniqueList[str] | None = pydantic.Field(
         default=None,
         description="The interfaces that the app can connect to.",
-        examples=["plugs: [home, removable-media]"],
+        examples=["[home, removable-media]"],
     )
+    """The list of interfaces that the app can connect to.
+
+    See the `content interface <https://snapcraft.io/docs/content-interface>`_
+    for more information about plugs and slots.
+    """
+
     aliases: UniqueList[str] | None = pydantic.Field(
         default=None,
-        description="The app's alternative internal identifiers.",
+        description="The aliases that can be used to run the app.",
         examples=["aliases: [my-app]"],
     )
+    """The aliases that can be used to run the app.
+
+    See `Commands and aliases <https://snapcraft.io/docs/commands-and-aliases>`_
+    for more information.
+    """
+
     environment: dict[str, str] | None = pydantic.Field(
         default=None,
-        description="The runtime environment variables available to the snap's apps.",
+        description="The runtime environment variables.",
         examples=[
-            "environment: {PYTHONPATH: $SNAP/usr/lib/python3/dist-packages, DISABLE_WAYLAND: 1"
+            "{PYTHONPATH: $SNAP/usr/lib/python3/dist-packages, DISABLE_WAYLAND: 1}"
         ],
     )
+    """The runtime environment variables.
+
+    To set an environment variable for all apps, use the top-level
+    ``environment`` key.
+    """
+
     command_chain: list[str] = pydantic.Field(
         default_factory=list,
-        description="The sequence of commands to run before the app's command runs. These commands also run when the user invokes ``snap run --shell``.",
-        examples=["command-chain: [bin/alsa-launch, bin/desktop-launch]"],
+        description="The ordered list of commands to run before the app's command runs.",
+        examples=["[bin/alsa-launch, bin/desktop-launch]"],
     )
+    """The ordered list of commands to run before the app's command
+    runs.
+
+    Command chains are useful to run setup scripts before running an app.
+    """
+
     sockets: dict[str, Socket] | None = pydantic.Field(
         default=None,
-        description="The app's sockets.",
-        examples=["listen-stream: $SNAP_COMMON/lxd/unix.socket, socket-mode: 0660"],
+        description="The sockets used to activate an app.",
+        examples=["$SNAP_COMMON/lxd/unix.socket, socket-mode: 0660"],
```

Comment:
Sockets have names, similar to apps:
```suggestion
        examples=["{my-socket: $SNAP_COMMON/lxd/unix.socket, socket-mode: 0660}"],
```

Reply:
https://github.com/canonical/snapcraft/pull/5421/commits/7cbca1d58702e90b48018f818a03a31f4d2cd1d9

Should `socket-mode` be a subkey of `my-socket` in this one? They're currently at the same indentation level.

Reply:
Oops yes, I misled you there.  It should render like this:
```yaml
    sockets:
      my-socket:
        listen-stream: "$SNAP_COMMON/lxd/unix.socket"
        socket-mode: 0660
```

Reply:
https://github.com/canonical/snapcraft/pull/5421/commits/f42b363320603021670ce9de364ee1644ff80508